R/three-legged-oauth.R

```diff
@@ -1,21 +1,35 @@
 
 #' Create a token for a user via the browser flow.
-#' @param scopes List of scopes required for the returned token.
-#' @param oauth_app `httr::oauth_app` to use for this token fetch. (Optional.)
+#' @param scopes A character vector of scopes to request.
+#' @param app An OAuth consumer application, created by [httr::oauth_app()].
+#' @param cache A logical value or a string. `TRUE` means to cache using the
+#'   default cache file, `.httr-oauth`. `FALSE` means don't cache, and `NA`
+#'   means to guess using some sensible heuristics. A string means use the
+#'   specified path as the cache file.
 #' @param ... Additional arguments (ignored)
 #' @export
-credentials_user_oauth2 <- function(scopes, oauth_app = NULL, ...) {
-
-  app <- httr::oauth_app(
-    "google",
-    key = "603366585132-nku3fbd298ma3925l12o2hq0cc1v8u11.apps.googleusercontent.com",
-    secret = "as_N12yfWLRL9RMz5nVpgCZt"
-  )
+#' @examples
+#' \dontrun{
+#' scope <- "https://www.googleapis.com/auth/drive"
+#' app <- httr::oauth_app(
+#'   appname = "my_awesome_app",
+#'   key = "keykeykeykeykeykey",
+#'   secret = "secretsecretsecret"
+#' )
+#'
+#' ## credentials_user_oauth2() is one of the functions token_fetch() will try
+#' token_fetch(scope, app = app)
+#' }
+credentials_user_oauth2 <- function(scopes,
+                                    app,
```

Comment:
Should we default to the tidyverse app?

Or default to another clearly-labeled-default app and let tidyverse libraries override? (I prefer this one.)

In particular, I think it's nice for users to be able to call `credentials_user_oauth2` by hand to kick the tires without first having to run the gauntlet of making/understanding client IDs. 😉